static/js/app.js

```diff
@@ -0,0 +1,215 @@
+  },
+
+  grabRoutes: function(){
+    $.get( "http://127.0.0.1:8080/routes", App.updateRoutes);
+  },
+
+  updateRoutes: function(data){
+    console.log(data);
+    var updatedRoute = [];
+    for(var i = 0; i < data.length; i ++){
+      console.log(data[i].name);
+      var points = [];
+      console.log(data[i].color)
+      for(var j = 0; j < data[i].coords.length; j ++){
+        points.push(new L.LatLng(data[i].coords[j].lat,data[i].coords[j].lng));
+      }
+      var polylineOptions = {
+        color: data[i].color,
+        weight: 3,//data[i]['width'],
+        opacity: 1
+      };
+      if(data[i].width === 0){
+        polylineOptions.dashArray = '10,10';
+      }
+
+      var polyline = new L.Polyline(points, polylineOptions);
+
+      var r ={
+        name: data[i].name,
+        id: data[i].id,
+        description: data[i].description,
+        color: data[i].color,
+        created: data[i].created,
+        enabled: data[i].enabled,
+        stops: data[i].stopsid,
+        start_time: data[i].startTime,
+        end_time: data[i].endTime,
+        points: points,
+        line: polyline
+      };
+
+      updatedRoute.push(r);
+
+    }
+
+    App.ShuttleRoutes = updatedRoute;
+    App.drawRoutes();
+
+  },
+
+
+  drawRoutes: function(){
+    for(i = 0; i < App.ShuttleRoutes.length; i ++){
+      App.ShuttleMap.removeLayer(App.ShuttleRoutes[i].line)
+    }
+    for(i = 0; i < App.ShuttleRoutes.length; i ++){
+      for(var j = 0; j < App.ShuttleRoutes[i].points.length; j ++){
+        App.MapBoundPoints.push(App.ShuttleRoutes[i].points[j]);
+      }
+    }
+
+    var polylineOptions = {
+      color: 'blue',
+      weight: 1,
+      opacity: 1
+    };
+    var polyline = new L.Polyline(App.MapBoundPoints, polylineOptions);
+    App.ShuttleMap.fitBounds(polyline.getBounds());
+    for(i = 0; i < App.ShuttleRoutes.length; i ++){
+      App.ShuttleMap.addLayer(App.ShuttleRoutes[i].line);
+    }
+
+  },
+
+  grabStops: function(){
+    $.get( "http://127.0.0.1:8080/stops", App.updateStops);
+
+  },
+
+  updateStops: function(data){
+    var stopIcon = L.icon({
+      iconUrl: 'static/images/stop.png',
+
+      iconSize:     [16, 16], // size of the icon
+      iconAnchor:   [0, 0], // point of the icon which will correspond to marker's location
+      shadowAnchor: [4, 62],  // the same for the shadow
+      popupAnchor:  [8, 8] // point from which the popup should open relative to the iconAnchor
+    });
+    for(var i = 0; i < data.length; i ++){
+      var stop = {
+        name: data[i].name,
+        description: data[i].description,
+        id: data[i].id,
+        latlng: [data[i].lat, data[i].lng],
+        marker: L.marker([data[i].lat,data[i].lng], {icon: stopIcon})
+      };
+      stop['marker'].bindPopup(stop.name);
+      stop['marker'].addTo(App.ShuttleMap).on('click', App.stopClicked);
+    }
+
+  },
+
+  grabVehicles: function(){
+    $.get( "http://127.0.0.1:8080/updates", App.updateVehicles);
```

Comment:
All of these GETs have to be relative (i.e. just "/updates") or it will break when deployed.